logql/v2/parser_test.go

```diff
@@ -451,6 +482,26 @@ func TestParseExpr(t *testing.T) {
 				},
 			},
 		},
+		{
+			input: `{first="value"} |~ "key\":\"val\""`,
+			expr: &LogQueryExpr{
+				filter: LogPipelineExpr{
+					&LogFilterExpr{
+						filter: "|~",
+						value:  "key\":\"val\"",
+					},
+				},
+				left: &StreamMatcherExpr{
+					matchers: []*labels.Matcher{
+						{
+							Type:  labels.MatchEqual,
+							Name:  "first",
+							Value: "value",
+						},
+					},
+				},
+			},
+		},
```

Comment:
What is the value added by this test case not covered by any of the existing?

Reply:
It's indeed similar to `TestQuotesEncode` below except this one tests double-quotes whereas the other one tests backquotes. I can move it to the `TestQuotesEncode` test, maybe it will be more clear & consistent